### PR TITLE
Fix TypeScriptRuleTest

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: its/plugin/projects/ts-rule-project/S4158.ts [hunk 1]
```diff
@@ -160,7 +160,7 @@ function arrayInitializedByFunctionCall(init: () => number[]) {
 
 function arrayNotInitialized() {
   let notInitializedArray!: number[];
-  foo(notInitializedArray[0]); // Noncompliant
+  foo(notInitializedArray[0]);  // Not reported
```
Comment (vilchik-elena) [inline its/plugin/projects/ts-rule-project/S4158.ts]: it's not reported in unit test as well

File: its/plugin/projects/ts-rule-project/S2871.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ arrayOfNumbers.sort(); // Noncompliant
 
 arrayOfNumbers.sort((n, m) => n - m);
 
-var emptyArrayOfNumbers: number[] = [];
+var emptyArrayOfNumbers: number[] = [1, 2, 3];
```
Comment (vilchik-elena) [inline its/plugin/projects/ts-rule-project/S2871.ts]: to not have issue for another rule

Comment (christophe-zurn-sonarsource) [reply its/plugin/projects/ts-rule-project/S2871.ts]: I would either remove this test or rename the variable, as it is confusing to have emptyArrayOfNumbers`` for a non-empty array

Comment (vilchik-elena) [reply its/plugin/projects/ts-rule-project/S2871.ts]: indeed I remove it

**Pull request merged**